### PR TITLE
DOCKER-81 Add support for monitoringPasscode passed as a secret and removal of in-clear-text livenessprobe httpheader

### PR DESCRIPTION
**Pull request by carminevassallo**
Currently, we cannot load the monitoringPasscode from an existing secret and we define it among liveness HTTP headers in clear text (DOCKER-65).

In this PR, we add the possibility to get this value from a secret and define a new liveness probe that masks the HTTP header.

### CODE REVIEW

**Comment by leo-geoffroy-sonarsource** [comment]
I think we should not create the default secret `{{ template "sonarqube.fullname" . }}-monitoring-passcode` when we provision monitoringPasscodeSecretName and monitoringPasscodeSecretKey.
It is failing on my side since `.Values.monitoringPasscode` is not defined

**Review by leo-geoffroy-sonarsource — CHANGES_REQUESTED** [review]
see comment above

**Comment by carminevassallo** [comment]
> I think we should not create the default secret `{{ template "sonarqube.fullname" . }}-monitoring-passcode` when we provision monitoringPasscodeSecretName and monitoringPasscodeSecretKey. It is failing on my side since `.Values.monitoringPasscode` is not defined

Hi @leo-geoffroy-sonarsource,

thanks I just have a question. Isn't `monitoringPasscode` always defined in the values.yml? By default, we always have this value set to a dummy password (I'm referring to [this line](https://github.com/SonarSource/helm-chart-sonarqube/blob/2a3bed2f63fee521223a3bc1e085d9baf583d8bd/charts/sonarqube/values.yaml#L299)), so it should not happen that `monitoringPasscode` is unset. Am I right? Could it be that you were creating a secret having the same name and this was causing the error?

File: charts/sonarqube-dce/values.yaml [hunk 1]
```diff
@@ -397,6 +397,9 @@ initFs:
 # a monitoring passcode needs to be defined in order to get reasonable probe results
 # not setting the monitoring passcode will result in a deployment that will never be ready
 monitoringPasscode: "define_it"
+# you can also define the passcode loading it from an existing secret specifying the right key
+# monitoringPasscodeSecretName: "passSecretName"
```
Comment (leo-geoffroy-sonarsource) [inline charts/sonarqube-dce/values.yaml:401]: maybe use a valid secret name as example

**Review by leo-geoffroy-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**